kedro/framework/cli/jupyter.py

```diff
@@ -90,87 +48,125 @@ def jupyter():
-    ip_address,
-    all_kernels,
     env,
-    idle_timeout,
     args,
     **kwargs,
-):  # pylint: disable=unused-argument,too-many-arguments
+):  # pylint: disable=unused-argument
     """Open Jupyter Notebook with project specific variables loaded."""
     _check_module_importable("jupyter_core")
 
     validate_settings()
 
-    if "-h" not in args and "--help" not in args:
-        ipython_message(all_kernels)
-
-    _update_ipython_dir(metadata.project_path)
-    arguments = _build_jupyter_command(
-        "notebook",
-        ip_address=ip_address,
-        all_kernels=all_kernels,
-        args=args,
-        idle_timeout=idle_timeout,
-        project_name=metadata.project_name,
-    )
+    kernel_name = f"kedro_{metadata.package_name}"
+    try:
+        _create_kernel(kernel_name, f"Kedro ({metadata.package_name})")
+    except Exception as exc:
+        raise KedroCliError(
+            f"Cannot setup kedro kernel for Jupyter.\nError: {exc}"
+        ) from exc
+
+    if env:
+        os.environ["KEDRO_ENV"] = env
 
-    python_call_kwargs = _build_jupyter_env(env)
-    python_call("jupyter", arguments, **python_call_kwargs)
+    python_call(
+        "jupyter",
+        ("notebook", f"--MappingKernelManager.default_kernel_name={kernel_name}")
+        + args,
+    )
 
 
 @forward_command(jupyter, "lab", forward_help=True)
-@click.option("--ip", "ip_address", type=str, default="127.0.0.1", help=JUPYTER_IP_HELP)
-@click.option(
-    "--all-kernels", is_flag=True, default=False, help=JUPYTER_ALL_KERNELS_HELP
-)
-@click.option("--idle-timeout", type=int, default=30, help=JUPYTER_IDLE_TIMEOUT_HELP)
 @env_option
 @click.pass_obj  # this will pass the metadata as first argument
 def jupyter_lab(
     metadata: ProjectMetadata,
-    ip_address,
-    all_kernels,
     env,
-    idle_timeout,
     args,
     **kwargs,
-):  # pylint: disable=unused-argument,too-many-arguments
+):  # pylint: disable=unused-argument
     """Open Jupyter Lab with project specific variables loaded."""
     _check_module_importable("jupyter_core")
 
     validate_settings()
 
-    if "-h" not in args and "--help" not in args:
-        ipython_message(all_kernels)
-
-    _update_ipython_dir(metadata.project_path)
-    arguments = _build_jupyter_command(
-        "lab",
-        ip_address=ip_address,
-        all_kernels=all_kernels,
-        args=args,
-        idle_timeout=idle_timeout,
-        project_name=metadata.project_name,
+    kernel_name = f"kedro_{metadata.package_name}"
+    try:
+        _create_kernel(kernel_name, f"Kedro ({metadata.package_name})")
+    except Exception as exc:
+        raise KedroCliError(
+            f"Cannot setup kedro kernel for Jupyter.\nError: {exc}"
+        ) from exc
+
+    if env:
+        os.environ["KEDRO_ENV"] = env
+
+    python_call(
+        "jupyter",
+        ("lab", f"--MultiKernelManager.default_kernel_name={kernel_name}") + args,
+    )
+
+
+def _create_kernel(kernel_name: str, display_name: str) -> None:
```

Comment:
This is the bit that might eventually become its own `kedro jupyter-init` or something like that.

I spent a loooong time reading through the ipython and jupyter source to try and figure out which were the right high-level API commands to use here to make this as robust as possible so we're not relying on too much stuff that might change. So🤞  this carries on working in the future, but I decided to the wrap the whole call in `try/except` just in case...